client/src/components/Alert.tsx

```diff
@@ -36,28 +30,34 @@ const statusIcon = {
   warning: <IoWarningOutline className='text-yellow-700' size={20} />,
 };
 
-export const Alert = ({ status, message }: AlertProp) => {
+export type AlertStatus = 'error' | 'success' | 'info' | 'warning';
+
+interface AlertProps {
+  status: AlertStatus;
+  message?: string;
+}
```

Comment:
What benefit do we get here whether its an interface, a type or an inline type? I think this PR could go through the components and be consistent with a format for props.

Reply:
You're right, I usually use a type (not interface) for props, it's good to be consistent. I like always extracting the prop type out of the function definition because it's easier to add new props when needed, and it's more readable compared to inline types.

Reply:
I changed all of the components to use a {ComponentName}Props type now